src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  };
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    const index = this.alive.findIndex(el => el === animal);
+
+    this.alive.splice(index, 1);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    this.type = 'herbivore';
+    Animal.addAnimal(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.type = 'carnivore';
+    Animal.addAnimal(this);
+  }
+
+  bite(victim) {
+    if (victim.type === 'herbivore' && victim.hidden === false) {
```

Comment:
small suggestion: use can just use NOT(!) operator instead of compare to false

Reply:
fixed... silly me  :)